packages/common-scripts/src/common.ts

```diff
@@ -1,142 +1,338 @@
 import {
   parseAddress,
-  generateAddress,
   TransactionSkeletonType,
   Options,
+  minimalCellCapacity,
 } from "@ckb-lumos/helpers";
-import secp256k1Blake160Multisig, {
-  FromInfo,
-} from "./secp256k1_blake160_multisig";
+import secp256k1Blake160Multisig from "./secp256k1_blake160_multisig";
+import { FromInfo, parseFromInfo } from "./from_info";
 import secp256k1Blake160 from "./secp256k1_blake160";
-import {
-  isSecp256k1Blake160Address,
-  isSecp256k1Blake160MultisigAddress,
-  prepareSigningEntries as _prepareSigningEntries,
-  isSecp256k1Blake160Script,
-  isSecp256k1Blake160MultisigScript,
-} from "./helper";
 import { getConfig, Config } from "@ckb-lumos/config-manager";
-import lockTimePool from "./locktime_pool";
-import { Address, Header } from "@ckb-lumos/base";
+import locktimePool from "./locktime_pool";
+import {
+  Address,
+  Header,
+  Script,
+  values,
+  Cell,
+  HexString,
+  Hash,
+  PackedSince,
+} from "@ckb-lumos/base";
+import anyoneCanPay from "./anyone_can_pay";
+const { ScriptValue } = values;
+import { Set, List } from "immutable";
+
+/**
+ * CellCollector should be a class which implement CellCollectorInterface.
+ */
+export interface LockScriptInfo {
+  code_hash: Hash;
+  hash_type: "type" | "data";
+  lockScriptInfo: {
+    CellCollector: any;
+    setupInputCell(
+      txSkeleton: TransactionSkeletonType,
+      inputCell: Cell,
+      fromInfo: FromInfo,
+      options: {
+        config?: Config;
+        defaultWitness?: HexString;
+        since?: PackedSince;
+        needCapacity?: HexString;
+      }
+    ): {
```

Comment:
Let's return a promise here for potentail usage

Reply:
updated.